### PR TITLE
Scenario additions, Achievement additions and Meta progression updates

### PR DESCRIPTION
**Pull request by RanikOrtega**
#### Summary
Content "More scenarios, achievements and EOC effects"

#### Purpose of change
More and varied scenarios, including under-utilized locations like Rooftops, Motels, Mobile Home parks etc etc. Addition of EOC events for scenario starts and improved effects. More Achievements for players to progress with and additional reason to explore the world. EX: "I'm going to try traveling to the aircraft carrier to complete the 'Top Gun' Achievement and unlock the Carrier Scenario

#### Describe the solution
Additional scenarios using remixed old content and some new EOC content. Location based achievements,  that lead to unlocking various existing challenge runs. 

#### Describe alternatives you've considered
None

#### Testing
Multiple tests for location based achievements. If they activated, displayed and unlocked scenarios via meta progression. Repeated spawning into new scenarios to make sure everything worked for the most part.

#### Additional context
This PR is partially dependent on https://github.com/CleverRaven/Cataclysm-DDA/issues/64762 as Scenario start EOC's that generate, Fire, Smoke and blood to much better effect than FIRE_START cause a seemingly harmless error on startup. 

CDDA-Fire Before
CDDA-Fire After
Scenario List New
CDDA-Fort
ntent.com/14862800/229667276-57f93fb1-47fa-4753-9d27-f3141006cd96.png)
CDDA-Island-Prison

More screenshots available if need be.

### CODE REVIEW

File: data/json/scenarios.json [hunk 1]
```diff
@@ -867,6 +875,139 @@
...
+    "points": -1,
+    "description": "You were trying to get some distance from home and keep a low profile. Days of boredom and a return to bad habits made you overdo it with the booze last night. You awake and find yourself confused with the strange behavior of people outside. Time to sober up and see what's going on. ",
+    "forced_traits": [ "ADDICTIVE" ],
+    "start_name": "In Town",
+    "allowed_locs": [ "sloc_mobile_home", "sloc_motel" ],
+    "eoc": [ "EOC_scenario_drunk" ],
+    "flags": [ "LONE_START" ]
   }
-]
+]
```
Comment (Maleclypse) [inline data/json/scenarios.json:1013]:
```suggestion
]

```

File: data/json/statistics.json [hunk 2]
```diff
@@ -2004,5 +2004,79 @@
...
+    }
+          ]
+  {
+    "id": "num_avatar_enters_hospital",
+    "type": "event_statistic",
+    "stat_type": "count",
+    "event_transformation": "avatar_enters_hospital",
+    "description": { "str": "hospital entered", "str_pl": "hospitals entered" }
   }
 ]
```
Comment (Maleclypse) [inline data/json/statistics.json:2082]:
```suggestion
]

```

File: data/json/statistics.json [hunk 3]
```diff
@@ -2004,5 +2004,80 @@
...
+            "urban_35_31",
+            "urban_35_32",
+            "urban_35_33",
+            "urban_35_34",
+            "urban_35_35"
+          ]
+        ]
+      }
+    }
+          ]
```
Comment (Maleclypse) [inline data/json/statistics.json:2074]:
```suggestion
          ]
      },
```

File: data/json/statistics.json [hunk 4]
```diff
@@ -2004,5 +2004,81 @@
...
```
Comment (Maleclypse) [inline data/json/statistics.json:2074]: Copy this entire file into https://jsonlint.com/ and then work through your brackets error here.  :)

Comment (RanikOrtega) [reply data/json/statistics.json:2074]: image

image


JSON Lint shows it as valid. The entire Statistics.json copied and pasted. Should I take additional steps?

**Comment by RanikOrtega** [comment]
At this point, Not sure what to do about the build errors. They seem unrelated to the .json files.

File: data/json/scenarios.json [hunk 5]
```diff
@@ -867,6 +877,139 @@
...
+    "eoc": [ "EOC_scenario_blood" ],
+    "map_extra": "mx_corpses",
+    "flags": [ "LONE_START" ]
+  },
+  {
+    "type": "scenario",
+    "id": "rock_hard_place_start",
+    "name": "Between a Rock and a Hard Place",
+    "points": -2,
+    "description": "You were trying to scavenge for supplies, now some rough group of bandits has dragged monsters in with them where you were hiding! ",
```
Comment (Brambor) [inline data/json/scenarios.json:969]:
```suggestion
    "description": "You were trying to scavenge for supplies, and now some rough group of bandits has dragged monsters in with them where you were hiding!",
```

Ment to fix only space, but Gramarly suggested adding "and", sounds better.

File: data/json/scenarios.json [hunk 6]
```diff
@@ -867,6 +877,139 @@
...
+    "map_extra": "mx_looters",
+    "requirement": "achievement_kill_10_monsters",
+    "flags": [ "LONE_START" ]
+  },
+  {
+    "type": "scenario",
+    "id": "shattered_sky_start",
+    "name": "Challenge - Shattered sky",
+    "points": -5,
+    "description": "From up here, you watched it all fall apart.  The sky cracked and indescribable alien shapes writhed in the clouds.  The horizon has returned to normal, but you still see shadows moving in the corner of your eye and feel them calling you. ",
```
Comment (Brambor) [inline data/json/scenarios.json:995]:
```suggestion
    "description": "From up here, you watched it all fall apart.  The sky cracked and indescribable alien shapes writhed in the clouds.  The horizon has returned to normal, but you still see shadows moving in the corner of your eye and feel them calling you.",
```

File: data/json/scenarios.json [hunk 7]
```diff
@@ -867,6 +877,139 @@
...
+    "eoc": [ "scenario_strong_portal_storm_base" ],
+    "requirement": "achievement_reach_mid_z_level",
+    "flags": [ "LONE_START", "CITY_START", "CHALLENGE", "HELI_CRASH" ]
+  },
+  {
+    "type": "scenario",
+    "id": "low_profile_start",
+    "name": "Low Profile",
+    "points": -1,
+    "description": "You were trying to get some distance from home and keep a low profile.  Days of boredom and a return to bad habits made you overdo it with the booze last night.  You awake and find yourself confused with the strange behavior of people outside.  Time to sober up and see what's going on. ",
```
Comment (Brambor) [inline data/json/scenarios.json:1008]:
```suggestion
    "description": "You were trying to get some distance from home and keep a low profile.  Days of boredom and a return to bad habits made you overdo it with the booze last night.  You awake and find yourself confused by the strange behaviour of people outside.  Time to sober up and see what's going on. ",
```

File: data/json/scenarios.json [hunk 8]
```diff
@@ -867,6 +877,139 @@
...
+      "sloc_river"
+    ],
+    "flags": [ "LONE_START", "HELI_CRASH" ]
+  },
+  {
+    "type": "scenario",
+    "id": "lone_survivor_start",
+    "name": "Lone Survivor",
+    "points": 0,
+    "description": "You are the only one that made it.  ",
```
Comment (Brambor) [inline data/json/scenarios.json:942]:
```suggestion
    "description": "You are the only one that made it.",
```

**Review by Brambor — CHANGES_REQUESTED** [review]
Fixing spaces at the end and changing the wording a bit (suggested by Gramarly).

File: data/json/scenarios.json (hunk 8 above)
Comment (RanikOrtega) [reply data/json/scenarios.json]: Added all recommended changes to updates Scenarios.json

**Comment by RanikOrtega** [comment]
Basic builds passed. Last two errors may be unrelated to PR. If I need to make any further changes let me know.

**Comment by RanikOrtega** [comment]
All tests passes except for Small Ubuntu error. Ready to merge?

**Comment by RanikOrtega** [comment]
All checks green. Ready to merge after review.

**Pull request merged**